### ISSUE TITLE
A `kotlin.collection.Map` passed as a parameter attributes into `NewRelic.recordBreadcrumb` will result in a runtime crash

### ISSUE THREAD

**Issue by jordanterry** [newrelic/newrelic-android-agent#275]
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description

When a `kotlin.collection.Map` (derived from a function that returns an immutable map) is passed as a parameter to `NewRelic.recordBreadcrumb` a runtime exception is thrown. 

[NOTE]: # ( Describe the problem you're encountering. )
[TIP]:  # ( Do NOT give us access or passwords to your New Relic account or API keys! )

### Populated Immutable Map

Code:

```kotlin
NewRelic.recordBreadcrumb(
    "Breadcrumb",
     mapOf("key" to "value"),
)
```

Runtime Exception:

```
Exception in thread "main" java.lang.UnsupportedOperationException
	at java.base/java.util.AbstractMap.put(AbstractMap.java:209)
	at com.newrelic.agent.android.NewRelic.recordBreadcrumb(NewRelic.java:943)
        ...
```

### Empty Immutable Map

Code:

```kotlin
NewRelic.recordBreadcrumb(
    "Breadcrumb",
    emptyMap()
)
```

Runtime Exception:

```
Exception in thread "main" java.lang.UnsupportedOperationException: Operation is not supported for read-only collection
	at kotlin.collections.EmptyMap.put(Maps.kt)
	at com.newrelic.agent.android.NewRelic.recordBreadcrumb(NewRelic.java:943)
        ...
```

### Using `mutableMapOf`

If I try to use `mutableMapOf` I see an error in my IDE. 

```kotlin
NewRelic.recordBreadcrumb(
    "Hi",
    mutableMapOf("key" to "value")
)
```
![Screenshot 2024-09-04 at 11 39 22](https://github.com/user-attachments/assets/3e1672ec-cfc4-4004-8239-3642fbef29d1)

[NOTE]: # ( Please be as specific as possible. )

## Expected Behavior

I should be able to use any `Map` without IDE issues or runtime exceptions when recording a breadcrumb.

## Relevant Logs / Console output

Logs provided above. 

## Your Environment

Kotlin + Android

**Comment by ndesai-newrelic** [comment]
@jordanterry 
**Current Limitation**
Our SDK is implemented in Java, which does not natively support mutable maps as method parameters. Due to this language constraint, we are unable to add methods that accept mutable maps as parameters.
**Planned Fix**
We acknowledge that this limitation has led to crashes when users attempt to pass mapOf() or emptyMap() as arguments. We want to assure you that we are actively working on a solution to address this issue.

**Comment by jordanterry** [comment]
@ndesai-newrelic would it be possible to create a mutable map internally? The fix doesn't necessarily have to be on the public API. 

Here a `HashMap` is created: 

https://github.com/newrelic/newrelic-android-agent/blob/8cec5e81e22b6fd529033e4991d4204d6e7973af/agent/src/main/java/com/newrelic/agent/android/NewRelic.java#L970-L975

When I get a free moment I can open a PR if that helps?

**Comment by ndesai-newrelic** [comment]
@jordanterry that's already done, it will release soon.

**Comment by ndesai-newrelic** [comment]
it is released in 7.6.1
